lib/trino-collect/src/test/java/io/trino/collect/cache/TestEvictableCache.java

```diff
@@ -108,6 +111,40 @@ public void testEvictByWeight()
         assertThat(cache.asMap().values().stream().mapToInt(String::length).sum()).as("values length sum").isLessThanOrEqualTo(20);
     }
 
+    @Test(timeOut = TEST_TIMEOUT_MILLIS)
```

Comment:
It looks guava cache allows for providing Ticker used a time source. You can use TestingTicker and make test not dependent on wall time.

Reply:
this was a really good suggestion